src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -120,22 +124,26 @@ public class AadIssuerValidator
 
             try
             {
-                if (securityToken.Issuer.EndsWith("v2.0", StringComparison.OrdinalIgnoreCase))
+                string AadIssuer;
+                if (validationParameters.ValidateIssuerWithLKG)
                 {
-                    if (AadIssuerV2 == null)
-                        AadIssuerV2 = CreateConfigManager(AadAuthorityV2).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
-
-                    if (IsValidIssuer(AadIssuerV2, tenantId, issuer))
-                        return issuer;
+                    AadIssuer = validationParameters.ConfigurationManager.LastKnownGoodConfiguration.Issuer;
                 }
                 else
                 {
-                    if (AadIssuerV1 == null)
-                        AadIssuerV1 = CreateConfigManager(AadAuthorityV1).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
-
-                    if (IsValidIssuer(AadIssuerV1, tenantId, issuer))
-                        return issuer;
+                    if (validationParameters.ConfigurationManager != null)
```

Comment:
We cannot use validationParameters.ConfigurationManager because the authority will be the one that user provided instead of v1/v2. so, we might not get the metadata we need to validate current token. That is why we were creating a configmanger for v1/v2 authorities.

Reply:
If ConfigurationManager has effective authority, this makes sense.